scripts/imgtool/image.py

```diff
@@ -585,12 +600,18 @@ def create(self, key, public_key_format, enckey, dependencies=None,
         # EC signatures so called Pure algorithm, designated to be run
         # over entire message is used with sha of image as message,
         # so, for example, in case of ED25519 we have here SHAxxx-ED25519-SHA512.
-        sha = hash_algorithm()
-        sha.update(self.payload)
-        digest = sha.digest()
-        message = digest;
-        tlv.add(hash_tlv, digest)
-        self.image_hash = digest
+        if not is_pure:
+            sha = hash_algorithm()
+            sha.update(self.payload)
+            digest = sha.digest()
+            message = digest
+            tlv.add(hash_tlv, digest)
+        else:
+            # Note that when Pure signature is used, hash TLV is not present.
```

Comment:
@nordicjm @nvlsianpu I am bypassing addition of hash TLV in case when signature is pure; pure signature is not using it, but maybe the hash may be useful for something else is worth leaving it as it is?

Reply:
@de-nordic Yes, it might be usefull: Hash is for check the integrity of the image, while the signature is for checking authenticity of the image. Usually HASH calculation is faster and doesn't require any key. I can expect that some cases might require just integrity calculation.

Reply:
Now the TLV sha is there, planning, for the future - not here, to add switch to not include it.